File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    removeDuplicates(nums) {
        let i=0;
        let set= new Set();
        let k=0;
        while (i<nums.length){
            if (set.has(nums[i])){
                i+=1;
                continue;
            }
            set.add(nums[i]);
            let temp=nums[k];
            nums[k]=nums[i];
            nums[i]=temp;
            k+=1;
            i+=1;
        }
        return k;
    }
}

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res=[]
        subset=[]
        
        def dfs(i):
            if i>=len(nums):
                # since python is dynamically typed, subset will always be modified, so we add a static copy of it instead of adding it directly
                res.append(subset.copy())
                return 
            #include the ith index number
            subset.append(nums[i])
            dfs(i+1)
            #don't include the ith index number
            subset.pop()
            dfs(i+1)
        
        dfs(0)
        return res